cmd/go-quai/config.go

```diff
@@ -0,0 +1,51 @@
+package main
+
+import (
+	"path/filepath"
+
+	"github.com/spf13/cobra"
+	"github.com/spf13/viper"
+
+	"github.com/dominant-strategies/go-quai/cmd/utils"
+	"github.com/dominant-strategies/go-quai/log"
+	"github.com/dominant-strategies/go-quai/params"
+)
+
+var configCmd = &cobra.Command{
+	Use:   "config",
+	Short: "creates the default config file",
+	Long: `creates the default config file in the location specified by the --config-dir flag. 
+			The default config file will contain all the default values for the flags.
+			Any flags passed in the command line here will also overwrite the default values in the config file.`,
+	RunE:                       runConfig,
+	SilenceUsage:               true,
+	SuggestionsMinimumDistance: 2,
+	Example:                    `go-quai start -log-level=debug`,
+	PreRunE:                    startCmdPreRun,
+}
+
+func init() {
+	rootCmd.AddCommand(configCmd)
+
+	for _, flagGroup := range utils.Flags {
+		for _, flag := range flagGroup {
+			utils.CreateAndBindFlag(flag, configCmd)
```

Comment:
Does `CreateAndBindFlag()` first check if the file/flag already exist? The vanilla mode of operation for this command should be to **not** overwrite existing configs. Perhaps we can give a `--force/-f` option if the user is ok with overwriting an existing config.

Reply:
Yes it doesn't overwrite, all its doing under the hood is reading the config and creating it if missing. It's the same behavior as running the node.